### PR TITLE
XSEDE_USER_TYPE to FEDERATED_USER_TYPE

### PR DESCRIPTION
**Pull request by ryanrath**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: html/internal_dashboard/js/admin_panel/SectionExistingUsers.js [hunk 1]
```diff
@@ -687,6 +676,7 @@ XDMoD.ExistingUsers = Ext.extend(Ext.Panel, {
         };
 
         self.resetDirtyState = function () {
+            settingsAreDirty = false;
```
Comment (jpwhite4) [inline html/internal_dashboard/js/admin_panel/SectionExistingUsers.js]: Should this change be in this pull request?

File: html/controllers/user_admin/update_user.php [hunk 2]
```diff
@@ -186,8 +186,8 @@ function ($carry, Acl $item) {
 
    $returnData['success'] = true;
 
-   $statusPrefix = $user_to_update->isXSEDEUser() ? 'XSEDE ' : '';
-   $displayUsername = $user_to_update->isXSEDEUser() ? $user_to_update->getXSEDEUsername() : $user_to_update->getUsername();
+   $statusPrefix = $user_to_update->isFederatedUser() ? 'Federated ' : '';
```
Comment (jpwhite4) [inline html/controllers/user_admin/update_user.php]: Will the displayUsername be correct now? you are changing the logic..

File: classes/NewRest/Controllers/UserControllerProvider.php [hunk 3]
```diff
@@ -131,7 +131,7 @@ private function extractUserData(XDUser $user)
             'first_name' => $user->getFirstName(),
             'last_name' => $user->getLastName(),
             'email_address' => $emailAddress,
-            'is_xsede_user' => $user->isXSEDEUser(),
+            'is_federated_user' => $user->isFederatedUser(),
```
Comment (smgallo) [inline classes/NewRest/Controllers/UserControllerProvider.php]: `is_federated_user` is already set on the following line. Why can't we just use that?

Comment (ryanrath) [reply classes/NewRest/Controllers/UserControllerProvider.php]: yeah, we can. I was in replace mode at that point. Removal incoming

File: html/controllers/user_admin/update_user.php (hunk 2 above)
Comment (ryanrath) [reply html/controllers/user_admin/update_user.php]: per our conversation: yes, the displayUsername will contain the correct value for Federated Users ( although the value will be different than before. This is expected. )

File: classes/XDUser.php [hunk 4]
```diff
@@ -2845,7 +2844,7 @@ public static function initializeXSEDEUser($username)
             $last_name
         );
 
-        $user->setUserType(XSEDE_USER_TYPE);                   // XSEDE User
+        $user->setUserType(FEDERATED_USER_TYPE);                   // XSEDE User
```
Comment (jpwhite4) [inline classes/XDUser.php]: Suggest remove the (now) inaccurate comment // XSEDE User

File: classes/XDUser.php [hunk 5]
```diff
@@ -2906,7 +2905,7 @@ public static function deriveUserFromXSEDEUser($username)
             "SELECT id FROM Users WHERE username=:username AND user_type=:user_type",
             array(
                 'username' => $username,
-                'user_type' => XSEDE_USER_TYPE
+                'user_type' => FEDERATED_USER_TYPE
```
Comment (jpwhite4) [inline classes/XDUser.php]: Rather than changing this variable I would just remove this (unused) function altogether (unless it is being used somewhere).

File: classes/XDUser.php [hunk 6]
```diff
@@ -2940,7 +2939,7 @@ public static function XSEDEUserExists($username)
             'SELECT id FROM moddb.Users WHERE username = :username AND user_type=:user_type',
             array(
                 'username' => $username,
-                'user_type' => XSEDE_USER_TYPE
+                'user_type' => FEDERATED_USER_TYPE
```
Comment (jpwhite4) [inline classes/XDUser.php]: This function needs to be gone too.

File: html/controllers/user_admin/delete_user.php [hunk 7]
```diff
@@ -24,8 +24,8 @@
        
       $username = $user_to_remove->getUsername();
    
-      $statusPrefix = $user_to_remove->isXSEDEUser() ? 'XSEDE ' : '';
-      $displayUsername = $user_to_remove->isXSEDEUser() ? $user_to_remove->getXSEDEUsername() : $user_to_remove->getUsername();
+      $statusPrefix = $user_to_remove->isFederatedUser() ? 'Federated ' : '';
```
Comment (jpwhite4) [inline html/controllers/user_admin/delete_user.php]: This line can go completely.

File: db/migrations/7.1.0-7.5.0/moddb.sql [hunk 8]
```diff
@@ -0,0 +1,2 @@
+-- Update the 'Federated' User Type color
+UPDATE UserTypes SET color = '#b914f6' WHERE type = 'Federated';
```
Comment (smgallo) [inline db/migrations/7.1.0-7.5.0/moddb.sql]: Need a newline here.

File: html/controllers/user_admin/update_user.php (hunk 2 above)
Comment (smgallo) [reply html/controllers/user_admin/update_user.php]: Prior to implementing Globus auth, the XSEDE oAuth service returned the username as "<username>;<display>" (e.g., "mchupa;Michael Chupa") so there was extra code needed to decode it for viewing. This is no longer necessary.

File: html/internal_dashboard/controllers/controller.php [hunk 9]
```diff
@@ -125,12 +125,10 @@
 
     case 'enum_user_types_and_roles':
 
-        $query = "SELECT id, type, color FROM moddb.UserTypes";
+        $query = "SELECT id, type, color FROM moddb";
```
Comment (smgallo) [inline html/internal_dashboard/controllers/controller.php]: Why did you remove the table from this select statement?

File: html/internal_dashboard/js/admin_panel/SectionExistingUsers.js [hunk 10]
```diff
@@ -210,7 +210,17 @@ XDMoD.ExistingUsers = Ext.extend(Ext.Panel, {
             root: 'user_types',
             autoLoad: true,
             baseParams: { operation: 'enum_user_types' },
-            fields: ['id', 'type']
+            fields: ['id', 'type'],
+            listeners: {
+                load: function (store, records) {
+                    for (var i = 0; i < records.length; i++) {
+                        var record = records[i];
+                        if (record.data.type === 'Federated') {
```
Comment (smgallo) [inline html/internal_dashboard/js/admin_panel/SectionExistingUsers.js]: Since we are already using `CCR.xdmod.FEDERATED_USER_TYPE` elsewhere why not use it here as well and check the id instead of the type string? If we need to make a change it's one less to make.

File: html/internal_dashboard/controllers/controller.php (hunk 9 above)
Comment (ryanrath) [reply html/internal_dashboard/controllers/controller.php]: Because I'm losing my mind

Comment (smgallo) [reply html/internal_dashboard/controllers/controller.php]: `s/losing/lost/` ? :)

Comment (ryanrath) [reply html/internal_dashboard/controllers/controller.php]: yup, pretty much
